src/software/ai/evaluation/calc_best_shot.h

```diff
@@ -10,6 +10,372 @@
 #include "software/world/team.h"
 #include "software/world/world.h"
 
+class AngleSegment
+{
+   public:
+    /**
+     * Constructs an AngleSegment represented by a top angle and bottom angle
+     *
+     * @param angle_top the most north angle
+     * @param angle_bottom the most south angle
+     */
+    AngleSegment(Angle angle_top, Angle angle_bottom)
+        : angle_top_(angle_top), angle_bottom_(angle_bottom)
```

Comment:
If this is a precondition to this class, it would be good to enforce it in the constructor.

Or alternatively, this could just be pass in 2 angles, and `getAngleTop()` could figure out/return the angle on top

Reply:
Was worried about the tradeoff of doing this vs performance and if it was worth it. Even though it's not that big of a performance hit, it's still something just for semantics. Not too sure which approach is better

Reply:
Imo, validation of args is nice because we know that we won't put the AngleSegment in a weird state, the performance hit from this should be negligible because I assume it would just be 

~~~
if (angle_top_ < angle_bottom_)
{
   throw std::invalid_argument("angle_top_ is below angle_bottom_"); 
}
~~~
We won't be throwing this exception at all, if things are working correctly.
So I think there mostly is value because it seems like a lot of this code depends on this ordering

Reply:
done